omero/sysadmins/unix/server-install-homebrew.rst

```diff
@@ -41,175 +51,280 @@ running::
-
-    :doc:`/developers/build-system`
+   :doc:`/developers/build-system`
         Developer documentation page on how to build the OMERO.server
 
+5. Once the build completes, the OMERO server build output will be located in :file:`~/Projects/Omero/code/openmicroscopy/dist`.
+   If it is not already open, open your :file:`.bash_profie`::
+
+    $ open -a TextEdit.app ~/.bash_profile
+
+   Prepend the :file:`bin` directory to your :envvar:`PATH`::
+
+    export PATH=~/Projects/Omero/code/openmicroscopy/dist/bin:$PATH
+
+   Using the command-line terminal, reload your :file:`.bash_profile` using::
+
+    $ source ~/.bash_profile
+
+   To ensure OMERO is correctly linked into your OS X :envvar:`PATH`, type the following in terminal and ensure
+   you get a similar output::
+
+    $ which omero
+    /Projects/omero/code/openmicroscopy/dist/bin/omero
+
+6. Activate the Virtualenv enviornment that we created earlier in the "Requirements"
+   section::
+
+    $ source ~/Virtual/Omero/bin/activate
+
+7. Install Python dependencies using pip::
+
+    $ pip install -r ~/Projects/omero/code/openmicroscopy/dist/share/web/requirements-py27-all.txt
+
+
+OMERO configuration
+-------------------
+
+1. From a fresh command-line terminal, start the database server::
+
+    $ pg_ctl -D /usr/local/var/postgres -l /usr/local/var/postgres/server.log -w start
+
+  .. note::
+   **(Optional)** To make life easier, you can add an ```alias``` to your :file:`.bash_profile`::
+    
+    # Start Virtualenv for OMERO
+    alias startVmOmero="source ~/Virtual/Omero/bin/activate"
+
+   You can also add an `alias` to start and stop the Postgres service::
 
+    alias startPg='pg_ctl -D /usr/local/var/postgres -l /usr/local/var/postgres/server.log -w start'
+    alias stopPg='pg_ctl -D /usr/local/var/postgres -l /usr/local/var/postgres/server.log -w stop'
 
-Then prepend the development :file:`bin` directory to your :envvar:`PATH` to
-pick the right executbale::
+   Reload :file:`.bash_profile` in OS X::
 
-    $ export PATH=~/code/projects/openmicroscopy/dist/bin:$PATH
+    $ source ~/.bash_profile
 
-and follow the steps for setting up the database and OMERO data directory as mentioned in the previous section.
+2. To use Omero, we need to first setup Postgres. Open a command-line terminal and run the 
+   following commands to create a user called *db_user* and database called *omero_database*::
+
+    $ createuser -w -D -R -S db_user
+    $ createdb -E UTF8 -O db_user omero_database
+
+3. Create directory for OMERO to store its data::
+
+    $ mkdir -p ~/Projects/Omero/data
+
+4. Start your Virtualenv enviornment we created earlier::
+
+    $ source ~/Virtual/omero/bin/activate
+
+5. Now set the OMERO configuration::
+
+    $ omero config set omero.data.dir ~/Projects/Omero/data
+    $ omero config set omero.db.name omero_database
+    $ omero config set omero.db.user db_user
+    $ omero config set omero.db.pass db_password
```

Comment:
Add lines 
```
bin/omero config set omero.web.application_server development
bin/omero config set omero.web.debug True
```
I do not see why we should instruct developers to use non-develpment web server, @sbesson @will-moore

Reply:
I think this is probably OK. This install page is not explicitly for developers (although it will be mostly used by devs) and there is the link to how to set up development server.

Reply:
> I do not see why we should instruct developers to use non-develpment web server, 

My 2cents (though not of high priority for this PR) is that we should be encouraging as _little_ use of Django's single-threaded development server as possible.

Reply:
Hmmm - I still use the Django dev server for all my work. Works for me (mostly)

Reply:
For one-off's, I certainly do start up the dev server now and then, but for having a feel for the reactivity of the application, it would be worth using nginx in front of OMERO.web frequently. If you see any particular barriers to being able to do so, it would be good to hear what they are.

Reply:
So, I just went to try running nginx locally, starting with looking at the docs, but I notice that we don't have them?! sysadmins/unix/server-install-homebrew.html page just says 
```
Basic setup for OMERO using NGINX:

$ export HTTPPORT=${HTTPPORT:-8080}
$ omero web config nginx-development --http $HTTPPORT > $(brew --prefix omero53)/etc/nginx.conf
See installation script step03_nginx.sh

For detailed instructions on how to deploy OMERO.web in a production environment such as NGINX please see OMERO.web administration.
```
But that links to sysadmins/unix/install-web.html which doesn't have ANY instructions on how to deploy web.
So that's my first barrier!

Reply:
Maybe leave it in then for this PR but plan to guide more toward the nginx way after 5.4.0?

Reply:
Considering all the work that took part on the OSX install/docs including _dropping_ `--with-nginx` that seems potentially negligent.

Reply:
For reference, here are the steps that I can use locally:

```
mv /usr/local/etc/nginx/nginx.conf /usr/local/etc/nginx/nginx.conf.orig
bin/omero web config nginx-development > /usr/local/etc/nginx/nginx.conf
nginx -t
nginx
bin/omero web start
```

Reply:
minimally those steps should be included
@will-moore will you be okay to open a new PR?

Reply:
To continue the "barriers to using nginx" conversation (maybe should be continued elsewhere), I find when developing OMERO.figure, I need each page refresh to load fresh ```figure.js``` from ```/Users/wmoore/Desktop/FIGURE/figure/omero_figure/static/figure/figure.js```, not from ```/Users/wmoore/Desktop/OMERO/openmicroscopy/dist/lib/python/omeroweb/static/figure/figure.js``` without having to ```$ bin/omero web restart``` each time.

Strangely I currently have
```
$ omero config get omero.web.application_server
wsgi-tcp
```
and run ``` $ omero web start``` in the background.
I can then test web at ```http://localhost:4080/``` **without** nginx running. I guess this must be using the Django development server?

Also, I like being able to add ```print``` statements in Django code and see them shown in the terminal when I run ```$ omeroweb/manage.py runserver```.

Reply:
>  I guess this must be using the Django development server?

Not exactly. Django has started multiple processes (as opposed to the development servers single) and you are accessing it directly. I've not tried this, but I'd assume it's minimally better than using the development server.

> Also, I like being able to add print statements in Django code and see them shown in the terminal when I run $ omeroweb/manage.py runserver.

I'd encourage more use of `logging`. Then you'd need to just `tail -f` the log file rather than watch the stdout of `runserver`.

Reply:
> To continue the "barriers to using nginx" conversation (maybe should be continued elsewhere)

Happy to reply on the appropriate document, card, or issue.

Reply:
continued at https://trello.com/c/VdQ4HAl6/65-using-nginx-in-development ....